sdk/servicebus/azure-servicebus/azure/servicebus/aio/_async_message.py

```diff
@@ -84,7 +86,13 @@ async def dead_letter(  # type: ignore
         """
         # pylint: disable=protected-access
         self._check_live(MESSAGE_DEAD_LETTER)
-        await self._settle_message(MESSAGE_DEAD_LETTER)
+
+        details = {
+            RECEIVER_LINK_DEAD_LETTER_REASON: reason,
+            RECEIVER_LINK_DEAD_LETTER_DESCRIPTION: description
+        }
+
+        await self._settle_message(MESSAGE_DEAD_LETTER, dead_letter_details=details)
```

Comment:
Do we need to update async _settle_via_mgmt_link?

Reply:
nope, `_settle_via_mgmt_link` is a helper method serving as wrapper of `receiver._settle_message`.